libs/ui/ui.jsx

```diff
@@ -389,32 +382,39 @@ export class TidyBlocksApp extends React.Component{
   }
 
   // Calls the file upload input.
-  loadWorkspaceClick(){
-    this.refs.workspaceFileUploader.click();
+  loadWorkspaceClick () {
+    this.refs.workspaceFileUploader.click()
   }
 
   // Updates the workspace after a file has been uploaded.
-  loadWorkspace(){
+  loadWorkspace () {
     const text = this.refs.workspaceFileUploader.files[0].text().then((text) => {
-      console.log(text)
+      const xml = Blockly.Xml.textToDom(text)
+      const workspace = this.getWorkspace().state.workspace
+      Blockly.Xml.clearWorkspaceAndLoadFromXml(xml, workspace)
     })
   }
 
   // Calls the file upload input.
-  loadCsvClick(){
-    this.refs.csvFileUploader.click();
+  loadCsvClick () {
+    this.refs.csvFileUploader.click()
   }
 
   // Processes and loads the csv after the file has been uploaded
-  loadCsv(){
-    const text = this.refs.csvFileUploader.files[0].text().then((text) => {
-      console.log(text)
+  loadCsv () {
+    const file = this.refs.csvFileUploader.files[0]
+    const name = file.name
+    file.text().then((text) => {
+      const label = name.replace('.csv', '')
+      const workspace = this.getWorkspace().state.workspace
+      const df = new DataFrame(csvToTable(text))
+      this.state.env.ui.userData.set(label, df)
```

Comment:
Probably needs to be a setState call to update, but I'll merge for now then adjust.